vpr/src/route/rr_graph2.cpp

```diff
@@ -1082,26 +1082,88 @@ static void load_block_rr_indices(const DeviceGrid& grid,
+                 *   BOTTOM-RIGHT  |  TOP & LEFT
+                 *   -------------------------------------------------------
+                 *   Other         |  First come first fit
+                 *   -------------------------------------------------------
+                 *
+                 * Special for IPINs:
+                 *   If there are multiple wanted sides, first come first fit is applied 
+                 *   This guarantee that there is only a unique rr_node 
+                 *   for the same input pin on multiple sides, and thus avoid multiple driver problems
+                 */
+                std::vector<e_side> wanted_sides;
+                if (grid.height() - 1 == y) { /* TOP side */
+                    wanted_sides.push_back(BOTTOM);
+                }
+                if (grid.width() - 1 == x) { /* RIGHT side */
+                    wanted_sides.push_back(LEFT);
+                }
+                if (0 == y) { /* BOTTOM side */
+                    wanted_sides.push_back(TOP);
+                }
+                if (0 == x) { /* LEFT side */
+                    wanted_sides.push_back(RIGHT);
+                }
+
+                /* If wanted sides is empty still, this block does not have specific wanted sides,
+                 * Deposit all the sides
+                 */
+                if (true == wanted_sides.empty()) {
+                    for (e_side side : {TOP, BOTTOM, LEFT, RIGHT}) {
+                        wanted_sides.push_back(side);
+                    }
+                }
+
                 //Assign indices for IPINs and OPINs at all offsets from root
                 for (int ipin = 0; ipin < type->num_pins; ++ipin) {
-                    for (int width_offset = 0; width_offset < type->width; ++width_offset) {
-                        int x_tile = x + width_offset;
-                        for (int height_offset = 0; height_offset < type->height; ++height_offset) {
-                            int y_tile = y + height_offset;
-                            for (e_side side : SIDES) {
+                    bool found_ipin_node = false;
+                    for (e_side side : wanted_sides) {
+                        for (int width_offset = 0; width_offset < type->width; ++width_offset) {
+                            int x_tile = x + width_offset;
+                            for (int height_offset = 0; height_offset < type->height; ++height_offset) {
+                                int y_tile = y + height_offset;
                                 if (type->pinloc[width_offset][height_offset][side][ipin]) {
                                     int iclass = type->pin_class[ipin];
                                     auto class_type = type->class_inf[iclass].type;
 
                                     if (class_type == DRIVER) {
                                         indices[OPIN][x_tile][y_tile][side].push_back(*index);
```

Comment:
This code should mirror the IPIN code; we should be making only one rr-node index for a given OPIN, regardless of how many sides or locations it appears on.  But we should store that inode in multiple spots so we connect edges to it properly (as in the IPIN discussion below).